Source/carthage/Archive.swift

```diff
@@ -78,31 +78,39 @@ public struct ArchiveCommand: CommandType {
 			return SignalProducer(values: Platform.supportedPlatforms)
 				.flatMap(.Merge) { platform -> SignalProducer<String, CarthageError> in
 					return SignalProducer(values: frameworks).map { framework in
-						return (options.directoryPath as NSString).stringByAppendingPathComponent((platform.relativePath as NSString).stringByAppendingPathComponent(framework))
+						return (platform.relativePath as NSString).stringByAppendingPathComponent(framework)
 					}
 				}
-				.filter { relativePath in NSFileManager.defaultManager().fileExistsAtPath(relativePath) }
+				.collect()
+				.flatMap(.Merge) { relativePaths -> SignalProducer<(relativePath: String, absolutePath: String), CarthageError> in
+					return SignalProducer(values: relativePaths).map { relativePath in
```

Comment:
You can move this map out of the `flatMap` and before the `collect`—that will make it quite a bit simpler.